fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -1098,6 +1155,28 @@ protected DateTime getCurrentDate(DateTime dateTime, TimeGrain timeGrain) {
     }
 
     /**
+     * Extract valid sort direction.
+     *
+     * @param columnWithDirection  Column and its sorting direction
+     *
+     * @return Sorting direction. If no direction provided then the default one will be DESC
+     */
+    protected SortDirection getSortDirection(List<String> columnWithDirection) {
+
+        SortDirection sortDirection;
+        try {
+            sortDirection = columnWithDirection.size() == 2 ?
+                    SortDirection.valueOf(columnWithDirection.get(1).toUpperCase(Locale.ENGLISH)) :
+                    SortDirection.DESC;
+        } catch (IllegalArgumentException ignored) {
+            String sortDirectionName = columnWithDirection.get(1);
```

Comment:
To pull this out, we need to make sure the size of the list should be two.  If there is only column in the list (Need to consider the default direction) then we need to have default value for the variable which suppose to contain the pulled string from columnWithDirection.get(1). 

We need to do this to make catch part happy. Rather then that, would like to keep as it is. 

    SortDirection sortDirection;                                                                      
    String requestedDirection = null;                                                                 
    try {                                                                                             
       if (columnWithDirection.size() == 2)     {                                                    
           requestedDirection = columnWithDirection.get(1);                                          
           sortDirection =  SortDirection.valueOf(requestedDirection.toUpperCase(Locale.ENGLISH));    
                                                                                                  
       }                                                                                             
       else {                                                                                        
           sortDirection = SortDirection.DESC;                                                       
       }                                                                                             
    } catch (IllegalArgumentException ignored) {                                                      
          LOG.debug(SORT_DIRECTION_INVALID.logFormat(requestedDirection));                              
        throw new BadApiRequestException(SORT_DIRECTION_INVALID.format(requestedDirection));          
    }                                                                                                 
    return sortDirection;